prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -417,8 +417,14 @@ public void run() {
                                         iterator.remove();
                                         continue;
                                     }
-                                    InstancesResult status = wfEngine.getStatus(entity, nominalTime,
-                                            new Date(nominalTime.getTime() + 200), PROCESS_LIFE_CYCLE, false);
+                                    InstancesResult status = null;
+                                    try {
+                                        status = wfEngine.getStatus(entity, nominalTime,
+                                                new Date(nominalTime.getTime() + 200), PROCESS_LIFE_CYCLE, false);
+                                    } catch (FalconException e) {
+                                        LOG.error("Unable to get status " + entity.getName() , e);
+                                        continue;
+                                    }
                                     if (status.getInstances().length > 0
```

Comment:
Are there chances that status remain null. If yes, then we should have a check for status is null or not.

Reply:
donot think so from the code path either it will throw exception or will return the status but to be on safer side will add the null check.